Update learning_tensorflow.py

zeynep68 opened the pull request.

Miffyli commented:
Hey, thanks for the PR! Could you write a bit of summary what has been done / what has been fixed? I assume this is to update the code to work with tensorflow 2.x?

zeynep68 commented:
Hi, yes you're right, this is an update to tensorflow 2.0.
I changed a little bit of the structure of the neural network (duel dqn) but the core is the same. Also I replaced the replay memory class by a deque.
Learning and choosing best action I outsourced to an agent class which consist the neural network as an instance variable.
Most of the code in main has been moved to the run method.

Miffyli commented:
LGTM, agent trains as expected and code looks good enough for a quick example! Thanks for the PR again :)
